apps/jobs/models.py

```diff
@@ -133,25 +133,30 @@ def save(self, *args, **kwargs):
                 logger.info("Submission is below for user {0} form participant_team {1} for challenge_phase {2}".format(
                     self.created_by.pk, self.participant_team.pk, self.challenge_phase.pk))
 
-            if hasattr(self.challenge_phase, 'max_submissions_per_day'):
-                submissions_done_today_count = Submission.objects.filter(
-                    challenge_phase__challenge=self.challenge_phase.challenge,
-                    participant_team=self.participant_team,
-                    challenge_phase=self.challenge_phase,
-                    # Get the midnight time of the day
-                    submitted_at__gte=timezone.now().replace(hour=0, minute=0, second=0, microsecond=0)).count()
-
-                failed_count = Submission.objects.filter(
-                    challenge_phase=self.challenge_phase,
-                    participant_team=self.participant_team,
-                    status=Submission.FAILED,
-                    # Get the midnight time of the day
-                    submitted_at__gte=timezone.now().replace(hour=0, minute=0, second=0, microsecond=0)).count()
-
-                if ((submissions_done_today_count + 1 - failed_count > self.challenge_phase.max_submissions_per_day) or
-                        (self.challenge_phase.max_submissions_per_day == 0)):
-                    logger.info("Permission Denied: The maximum number of submission for today has been reached")
-                    raise PermissionDenied({'error': 'The maximum number of submission for today has been reached'})
+            submissions_done_today = Submission.objects.filter(
```

Comment:
You can be smart about making queries to the database. First, pick the common filter attributes and store them in a query set. Then use that queryset to filter out based on other uncommon attributes.